starlite/utils/extractors.py

```diff
@@ -0,0 +1,416 @@
+    def extract_path(connection: "ASGIConnection[Any, Any, Any]") -> str:
+        """
+
+        Args:
+            connection: An [ASGIConnection][starlite.connection.ASGIConnection] instance.
+
+        Returns:
+            The connection's scope["path"] value
+        """
+        return connection.scope["path"]
+
+    def extract_headers(self, connection: "ASGIConnection[Any, Any, Any]") -> Dict[str, str]:
+        """
+
+        Args:
+            connection: An [ASGIConnection][starlite.connection.ASGIConnection] instance.
+
+        Returns:
+            A dictionary with the connection's headers.
+        """
+        return obfuscate(
+            {k.decode("latin-1"): v.decode("latin-1") for k, v in connection.scope["headers"]}, self.obfuscate_headers
+        )
+
+    def extract_cookies(self, connection: "ASGIConnection[Any, Any, Any]") -> Dict[str, str]:
+        """
+
+        Args:
+            connection: An [ASGIConnection][starlite.connection.ASGIConnection] instance.
+
+        Returns:
+            A dictionary with the connection's cookies.
+        """
+        return obfuscate(connection.cookies, self.obfuscate_cookies)
+
+    def extract_query(self, connection: "ASGIConnection[Any, Any, Any]") -> Any:
+        """
+
+        Args:
+            connection: An [ASGIConnection][starlite.connection.ASGIConnection] instance.
+
+        Returns:
+            Either a dictionary with the connection's parsed query string or the raw query byte-string.
+        """
+        return connection.query_params if self.parse_query else connection.scope.get("query_string", b"")
+
+    @staticmethod
+    def extract_path_params(connection: "ASGIConnection[Any, Any, Any]") -> Dict[str, Any]:
+        """
+
+        Args:
+            connection: An [ASGIConnection][starlite.connection.ASGIConnection] instance.
+
+        Returns:
+            A dictionary with the connection's path parameters.
+        """
+        return connection.path_params
+
+    @staticmethod
+    def extract_method(request: "Request[Any, Any]") -> "Method":
+        """
+
+        Args:
+            request: A [Request][starlite.connection.Request] instance.
+
+        Returns:
+            The request's scope["method"] value.
+        """
+        return request.scope["method"]
+
+    @staticmethod
+    def extract_content_type(request: "Request[Any, Any]") -> Tuple[str, Dict[str, str]]:
+        """
+
+        Args:
+            request: A [Request][starlite.connection.Request] instance.
+
+        Returns:
+            A tuple containing the request's parsed 'Content-Type' header.
+        """
+        return request.content_type
+
+    async def extract_body(self, request: "Request[Any, Any]") -> Any:
+        """
+
+        Args:
+            request: A [Request][starlite.connection.Request] instance.
+
+        Returns:
+            Either the parsed request body or the raw byte-string.
+        """
+        if request.method != HttpMethod.GET:
```

Comment:
Flipping this with an early return would reduce lots of nesting.

Also, there is nothing stopping someone from making a GET request with a body (even though it's technically not allowed). Obviously we don't want to encourage this, but we may wish to support something that is technically possible?

Reply:
fair point.